src/Core/Foundation/IoC/Container.php

```diff
@@ -104,9 +104,8 @@ private function makeInstanceFromClassName($className, array $alreadySeen)
 
         if ($classConstructor) {
             foreach ($classConstructor->getParameters() as $param) {
-                $paramClass = $param->getClass();
-                if ($paramClass) {
-                    $args[] = $this->doMake($param->getClass()->getName(), $alreadySeen);
+                if ($param->getType() && !$param->getType()->isBuiltin()) {
```

Comment:
```
$param->getType()->isBuiltin()
```
this only works with ReflectionNamedType
https://www.php.net/manual/en/reflectionnamedtype.isbuiltin.php

Are we sure `$param->getType()` is a ReflectionNamedType? If yes I think PHPStan will require us to type-hint it